linenoise.c

```diff
@@ -1168,7 +1170,11 @@ int linenoiseHistorySave(const char *filename) {
     fp = fopen(filename,"w");
     umask(old_umask);
     if (fp == NULL) return -1;
-    chmod(filename,S_IRUSR|S_IWUSR);
+    if (chmod(filename,S_IRUSR|S_IWUSR) < 0) {
```

Comment:
Admittedly, I've paid zero attention to the larger context but...

fopen(..., "w") creates a new file... it's possible that the file doesn't exist prior to the fopen

I'm guessing that the intent of this snippet is to create a file and make it read/write only for the creator (0600)... what the author is protecting against, I don't know.